flocker/node/agents/gce.py

```diff
@@ -448,44 +488,58 @@ def create_volume(self, dataset_id, size):
         )
 
     def attach_volume(self, blockdevice_id, attach_to):
-        config = dict(
-            deviceName=blockdevice_id,
-            autoDelete=False,
-            boot=False,
-            source=(
-                "https://www.googleapis.com/compute/v1/projects/%s/zones/%s/"
-                "disks/%s" % (self._project, self._zone, blockdevice_id)
+        with start_action(
+                action_type=u"flocker:node:agents:gce:attach_volume",
+                blockdevice_id=blockdevice_id,
+                attach_to=attach_to,
+        ) as action:
+            config = dict(
+                deviceName=blockdevice_id,
+                autoDelete=False,
+                boot=False,
+                source=(
+                    "https://www.googleapis.com/compute/v1/projects/%s/zones/"
+                    "%s/disks/%s" % (self._project, self._zone, blockdevice_id)
+                )
             )
-        )
-        try:
-            # TODO(mewert): Verify timeout and error conditions.
-            # TODO(mewert): Test what happens when disk is attached RW to a
-            #               different instance, raise the correct error.
-            result = self._do_blocking_operation(
-                self._compute.instances().attachDisk,
-                instance=attach_to,
-                body=config
+            try:
+                # TODO(mewert): Verify timeout and error conditions.
+                # TODO(mewert): Test what happens when disk is attached RW to a
+                #               different instance, raise the correct error.
+                result = self._do_blocking_operation(
+                    self._compute.instances().attachDisk,
+                    instance=attach_to,
+                    body=config
+                )
+            except HttpError as e:
```

Comment:
:question: It might be easier to answer the question below if we always logged a traceback here. If the deployer is working correctly, this shouldn't normally happen, I think.